Bottom: b4a58036cc904128f82282b3197cb29c57ce026d
Top:    6bbdc8b080f7b0eedc3d140f32839a34a4f6f859
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-21 00:09:12 +0100

Add profiling to file load

















---

diff --git a/src/file.c b/src/file.c
index b6c67b9..018b272 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
